### PR TITLE
Fix for test script NewScriptEventButton_HappyPath_ContainsSCCategory

### PR DESCRIPTION
**Pull request by michleza**
adjusted logic in step 6, which caused flaky test behavior
replaced a bunch of strings with constants
reorganized the script a bit to be more readable.

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
 
 from PySide2 import QtWidgets
-import azlmbr.legacy.general as general
-
-import editor_python_test_tools.pyside_utils as pyside_utils
 from editor_python_test_tools.utils import TestHelper as helper
 from editor_python_test_tools.utils import Report
+import azlmbr.legacy.general as general
+import editor_python_test_tools.pyside_utils as pyside_utils
+from scripting_utils.scripting_constants import WAIT_TIME_3, SCRIPT_CANVAS_UI,ASSET_EDITOR_UI, NODE_PALETTE_QT
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py:13]: space between each import entry for consistency

File: AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py [hunk 2]
```diff
@@ -53,18 +49,19 @@ class TestAssetEditor_NewScriptEvent:
     async def run_test(self):
         # 1) Open Script Canvas window (Tools > Script Canvas)
         general.idle_enable(True)
-        general.open_pane("Script Canvas")
-        helper.wait_for_condition(lambda: general.is_pane_visible("Script Canvas"), 5.0)
+        general.open_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py:53]: Is there an issue with reducing the wait time here? What's been identified as the root cause of the test failure here? Are these standard workflows that we're using and correcting bad test consistency?

File: AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py [hunk 3]
```diff
@@ -53,18 +49,19 @@ class TestAssetEditor_NewScriptEvent:
...
 
         # 2) Close any existing AssetEditor window
-        general.close_pane("Asset Editor")
-        helper.wait_for_condition(lambda: not general.is_pane_visible("Asset Editor"), 5.0)
+        general.close_pane(ASSET_EDITOR_UI)
+        helper.wait_for_condition(lambda: not general.is_pane_visible(ASSET_EDITOR_UI), WAIT_TIME_3)
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py:57]: Is there an issue with reducing the wait time here? What's been identified as the root cause of the test failure here? Are these standard workflows that we're using and correcting bad test consistency?

File: AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py [hunk 4]
```diff
@@ -53,18 +49,19 @@ class TestAssetEditor_NewScriptEvent:
...
 
         # 3) Get the SC window object
         editor_window = pyside_utils.get_editor_main_window()
-        sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-        node_palette = sc.findChild(QtWidgets.QDockWidget, "NodePalette")
+        sc = editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
+        node_palette = sc.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
         frame = node_palette.findChild(QtWidgets.QFrame, "searchCustomization")
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py:63]: Is this a one-off node pallet child or should we be constant parameterizing this value as well?

File: AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py [hunk 5]
```diff
@@ -19,10 +19,6 @@ class Tests:
     new_asset = "Asset Editor created with new asset"
     script_event = "New Script event created in Asset Editor"
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NewScriptEventButton_HappyPath_ContainsSCCategory.py:20]: Previous tests have used the the `var = ('pass condition message', 'fail condition message')`, why is that pattern not being used here?

**Review by FuzzyCarterAWS — CHANGES_REQUESTED** [review]

**Review by FuzzyCarterAWS — APPROVED** [review]
LGTM

**Review by lsemp3d — APPROVED** [review]

**Pull request merged**